lib/charms/legend_database_manager/v0/database_manager.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2021 Canonical
+# See LICENSE file for licensing details.
+
+import json
+
+from ops import framework
+
+
+# The unique Charmhub library identifier, never change it
+LIBID = "431732f8afb641a3a5a38e5c5d01ee11"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 0
+
+REQUIRED_LEGEND_DATABASE_CREDENTIALS = [
+    "username", "password", "database", "uri"]
+
+
+class LegendDatabaseConsumer(framework.Object):
+    def __init__(self, charm, relation_name="legend-db"):
+        super().__init__(charm, relation_name)
+        self.charm = charm
+        self.relation_name = relation_name
+
+    def database_connection_from_mongodb(
```

Comment:
Suggestion: If the purpose of this method is to return information about database currently available would you prefer a name such as `database_info()` instead ?
Also I am not sure why `MongoConsumer` is an argument to this method instead of a member of an instance of this class.

Reply:
The purpose of this method is to "extract" the mongo credentials in the standardized format returned by this consumer. (and thus the other charms which will use it)

On the class-member-vs-parameter point: the main purpose of this library class is to be used in the "client charms" (i.e. SDLC, Engine, and Studio) so those would neither use this method, nor have access to a `MongoConsumer` instance to instantiate this class with.
This particular method is only used in the `_on_db_relation_joined` of the Legend DB Manager charm (NOTE: the relation between it and Mongo, not the relation between it and the other Legend components, which don't comply to the same `mongodb_datastore` interface)